swift/ql/src/queries/Security/CWE-328/WeakPasswordHashing.qhelp

```diff
@@ -0,0 +1,96 @@
+<!DOCTYPE qhelp PUBLIC
+        "-//Semmle//qhelp//EN"
+        "qhelp.dtd">
+<qhelp>
+    <overview>
+        <p>
+            Using a insufficiently computationally hard hash function can leave data
+            vulnerable, and should not be used for password hashing.
+        </p>
+
+        <p>
+            A strong cryptographic hash function should be resistant to:
+        </p>
+        <ul>
+            <li>
+                <strong>Pre-image attacks</strong>. If you know a hash value <code>h(x)</code>,
+                you should not be able to easily find the input <code>x</code>.
+            </li>
+            <li>
+                <strong>Collision attacks</strong>. If you know a hash value <code>h(x)</code>,
+                you should not be able to easily find a different input
+                <code>y</code>
+                with the same hash value <code>h(x) = h(y)</code>.
+            </li>
+            <li>
+                <strong>Brute force</strong>. If you know a hash value <code>h(x)</code>,
+                you should not be able to find an input <code>y</code> that computes to that hash value
+                using brute force attacks without significant computational effort.
+            </li>
+        </ul>
+
+        <p>
+            All of MD5, SHA-1, SHA-2 and SHA-3 are weak against offline brute forcing, since they are not computationally hard. This includes SHA-224, SHA-256, SHA-384 and SHA-512, which are in the SHA-2 family.
+        </p>
+
+        <p>
+            Password hashing algorithms are designed to be slow and/or memory intenstive to compute, which makes brute force attacks more difficult.
+        </p>
```

Comment:
```suggestion
        <p>
            Password hashing algorithms should be slow and/or memory intensive to compute, to make brute force attacks more difficult.
        </p>
```

One part of this is just a typo; the rest is up to you whether to accept as-is or modify. I only suggested the "should be" because I think it matches slightly better with the prior recommendation `You should not use such functions for password hashing`

Reply:
I prefer your new phrasing. :)